rpc/src/lib.rs

```diff
@@ -198,20 +207,72 @@ impl subsystem::Subsystem for Rpc {
     }
 }
 
-/// Construct a basic authorization header for HTTP requests (when [username_password] is set)
-pub fn make_http_header_with_auth(username_password: Option<(&str, &str)>) -> http::HeaderMap {
+#[derive(Clone)]
+pub enum RpcAuthType {
```

Comment:
I would call this RpcAuthData. I would use "Type" if this would be a numeric enumeration, not containing auth data.